### PR TITLE
migrate components to pure & connected

### PR DESCRIPTION
**Pull request by rgbkrk**
Moving everything from `providers` into `components`, bit by bit.

* [x] cell creator
* [x] toolbar
* [x] editor
* [x] notebook app

I ran out of laptop battery at the coffee shop so I'm going to skip `<Styles />` for now. Amusingly, it's the only actual provider. 😅

### CODE REVIEW

File: applications/jupyter-extension/nteract_on_jupyter/app/index.js [hunk 1]
```diff
@@ -10,7 +10,13 @@ import configureStore from "./store";
...
 import type { JupyterConfigData } from "./store";
 
-import { NotebookApp, Styles } from "@nteract/core/providers";
+import {
+  NotebookApp,
+  Styles,
+  ModalController,
+  NotebookMenu,
+  TitleBar
+} from "@nteract/core";
```
Comment (theengineear) [inline applications/jupyter-extension/nteract_on_jupyter/app/index.js:19]: 💯

File: packages/core/src/components/notebook-app.js [hunk 2]
```diff
@@ -13,29 +13,26 @@ import {
...
 
 import { Display, RichestMime } from "@nteract/display-area";
 
-import CodeMirror from "./editor";
+import Editor from "./editor";
 
 import { displayOrder, transforms } from "@nteract/transforms";
 
-import { Input, Prompt, Source, Pagers, Outputs, Cell } from "../components";
+import { Input, Prompt, Source, Pagers, Outputs, Cell } from "./presentational";
```
Comment (theengineear) [inline packages/core/src/components/notebook-app.js:29]: Cool. The `/components/presentational` works for me.

Comment (theengineear) [reply packages/core/src/components/notebook-app.js:29]: The idea is that presentational components don't need connecting or anything right? They're like dumb components?

File: packages/core/src/providers/index.js [hunk 3]
```diff
@@ -1,8 +1,4 @@
 // @flow
-import CellCreator from "./cell-creator";
-import Editor from "./editor";
-import NotebookApp from "./notebook-app";
 import Styles from "./styles";
-import Toolbar from "./toolbar";
 
-export { CellCreator, Editor, NotebookApp, Styles, Toolbar };
+export { Styles };
```
Comment (theengineear) [inline packages/core/src/providers/index.js:4]: 🐮 just `export { Styles } from "./styles";`

**Review by theengineear — APPROVED** [review]
Cool! Looks great! Sorry it took so long to look at. I'm going to merge this in and you can respond to comments at your leisure.

File: packages/core/src/components/notebook-app.js (hunk 2 above)
Comment (rgbkrk) [reply packages/core/src/components/notebook-app.js:29]: Yeah they're basically just styled divs that take certain kinds of children. Open to naming on these. I just moved them out of index.js so I wouldn't get a circular dependency.

File: packages/core/src/providers/index.js (hunk 3 above)
Comment (rgbkrk) [reply packages/core/src/providers/index.js:4]: I'll make the next PR that just moves this into the components.

**Pull request merged**